### ISSUE TITLE
Add support for changing options

### ISSUE THREAD

**Issue by pieterjandesmedt** [paradox41/angular-chartist.js#36]
I noticed the chart doesn't redraw when changing the options (for example switching from a stacked to an unstacked bar chart through the `stackBars` option). 

I modified `AngularChartistCtrl.prototype.renderChart` into
```javascript
    AngularChartistCtrl.prototype.renderChart = function (element, chartType, chartOptions) {
        if (chartType) {
            this.chartType = chartType;
        }
        if (chartOptions) {
            this.chartOptions = chartOptions;
        }

        return Chartist[this.chartType](element, this.data, this.options, this.responsiveOptions);
    };
```
and the watch into 
```javascript
                scope.$watch(function () {
                    return {
                        data: scope.data(),
                        chartType: scope.chartType,
                        chartOptions: scope.chartOptions()
                    };
                }, function (newConfig, oldConfig) {
                    console.log('data changed into %o', newConfig);
                    var newData = newConfig.data;
                    var oldData = oldConfig.data;

                    var newChartType = newConfig.chartType;
                    var oldChartType = oldConfig.chartType;

                    var newChartOptions = newConfig.chartOptions;
                    var oldChartOptions = oldConfig.chartOptions;

                    if (newData !== oldData) {
                        chart.update(newData);
                    }

                    if (newChartType !== oldChartType || newChartOptions !== oldChartOptions) {
                        chart = Ctrl.renderChart(elm, newChartType, newChartOptions);
                    }

                }, true);
```

but now this error occurs sometimes:
```
TypeError: Cannot read property 'currentOptions' of undefined
    at Line.update (chartist.js:1085)
    at Object.fn (angular-chartist.js:80) —> this is chart.update(newData);
    at Scope.$get.Scope.$digest (angular.js:14225)
    at Scope.$get.Scope.$apply (angular.js:14488)
```
(which incidentally made me notice — because of the line numbers — that installing angular-chartist.js via npm creates a dependency on chartist 0.5.0 instead of the current 0.7.1)

Do you have any thoughts as to why this happens?

**Comment by paradox41** [comment]
Not sure...There is an outstanding [PR](https://github.com/paradox41/angular-chartist.js/pull/33) that resolves the first issue you are discussing. I asked that it be updated to follow conventions for this repo with no response as of yet. If you are interested in making a PR with the fix, go for it. It is as simple as you made it in your example.

As to the random error, it could just be that it is an out of date dependency. Feel free to make a separate PR for that if you'd like. Otherwise, I will try and get to it this weekend.

**Comment by pieterjandesmedt** [comment]
The errors don't seem to crop up when updating the chart instead of rerendering it, as shown in @rolandschuetz' solution to this problem. I created a PR #37.

**Comment by paradox41** [comment]
Closed via 3cebded66f73506aa42ea0707a94d556d2cb0662
